src/models/machine_learning/MLmodel.jl

```diff
@@ -65,20 +68,45 @@ Creates a new feed-forward neural network.
 - `architecture`: `Flux.Chain` neural network architecture (optional)
 - `θ`: Neural network parameters (optional)
 """
-function NeuralNetwork(params::P;
-            architecture::Union{ChainType, Nothing} = nothing,
-            θ::Union{ComponentArrayType, Nothing} = nothing) where {P <: Sleipnir.Parameters, ChainType <: Lux.Chain, ComponentArrayType <: ComponentArray}
+function NeuralNetwork(
```

Comment:
I think we should also discuss if we should specify our own machine learning models, or we should completely externalize them. I think it's still a good idea to provide some "default" NNs of different sizes, since for many tasks the type of MLP will be the same. 

And as for my previous comment, in #302 we came up with a different way to "connect" the regressor with a target, inside a `Law` type. I think that perhaps your `Target` type should go into the `Law` type, basically constructing the internals of whatever high-level decision has been made through the `Law` interface.

Reply:
Right now I am allowing the two options, and I think we should keep it like that: either provide the architecture and we handle the setup of the NN, or don't provide the architecture and just build the default NN. We do want to be able to design the architecture, since this is one of the tools the modeler would like to explore and we want to be as flexible as possible with this, including different architectures, activation functions, and initializations of the weights. Allowing to pass the Lux.Chain is the perfect compromise between these, which is what I currently did in one of the last commits.